### ISSUE TITLE
Dangerously misleading section about migration in manual

### ISSUE THREAD

**Issue by wucke13** [mailcow/mailcow-dockerized#2591]
What is written [here](https://mailcow.github.io/mailcow-dockerized-docs/i_u_m_migration/) IMHO is very bad. It destroys existing volumes of other docker containers on the target (as it overwrites `/var/lib/docker/volumes/metadata.db`). Why not using the nice `helper-scripts/backup_and_restore.sh` script for this purpose? I would suggest the following:

+ download mailcow-dockerized on new system
+ mount folder from new system on old system via sshfs
+ full backup with helper script to the mount
+ copy `mailcow.conf` to new systems mailcow folder
+ once the backup is finished, use the restore functionality to restore mailcow-dockerized from the backup

**Comment by hachre** [comment]
The guide is meant for transitioning to a brand new server that doesn't even have Docker installed yet. But yes, **we should at least add an additional warning at Step 3 that mentions that it will delete any other existing Docker stuff on the target machine**. 

I haven't tested your proposed method, but the main drawback I see here is, it would take a lot more downtime to do it this way. If I remember correctly the bug discussion that spawned this entry in the guide depended on a fast transition with low downtime.

**Comment by wucke13** [comment]
Ah I see (though that is not explicitly stated in the guid either). Though, there is something which may be done about it: first of all, the creation of the tar archives and compression using gzip is very slow (please have a look at #1369 ), but can be tuned a lot. Depending on version, schily tar out performs gnutar by a huge margin. Furthermore at least `pigz` can be used to speed up `gzip`. Another Algorithm like `lz4` or `bzip2`, which natively supports parallel compression and decompression, may be used. Just the particular combination of gnutar and normal gzip is really slow. Using the `sshfs` mount further reduces the necessary downtime. 

Secondly when changing hosts, the biggest downtime will originate from the fact, that updating the DNS records takes some time - in my case well over 10 Hours until all mail servers I used to test the new setup knew about the new ip address.

Third: for the `vmail` volume, a special option may be added to merge existing files and folders with those of the backup. This way I can omit initially copying the `vmail` volume. With blocked ports for IMAP (so that users don't mess up) the transition can be made really fast (as the most time is spent archiving the `vmail` folder), and new Mails both on new and old host may arrive. After a few hours, the `vmail` folder from the old host finally may be merged into the new hosts `vmail` folder.

**Comment by andryyy** [comment]
Feel free to create a PR. There are several ways to migrate, you can add new ways to the docs any time.

I think the given docs describe a very basic migration for small environments to a new server. Your described methods will also not be the best or apply to all setups. There is no "best way" for all setups in general.

There are so many more things to consider before migrating larger environments anyway...

This issue should be closed with a doc update. If you feel like it, go ahead, @hachre

**Comment by hachre** [comment]
I've made the required changes to the Documentation in https://github.com/mailcow/mailcow-dockerized-docs/commit/d07c241f21d3e84fbcf021ebe4457a8ba71075fa

**Comment by wucke13** [comment]
@hachre Maybe it's worth mentioning that starting a new Mailcow instance from a copy of the old mailcow.conf and then using backup/restore to migrate the data works too

**Comment by hachre** [comment]
I've now added a small section mentioning this alternative migration method.

**Comment by wucke13** [comment]
Nice!

**Comment by hachre** [comment]
Sorry, @andryyy, just saw your comment from yesterday now:
https://github.com/mailcow/mailcow-dockerized-docs/commit/d07c241f21d3e84fbcf021ebe4457a8ba71075fa#commitcomment-33528968

Will do PRs from now on 👍
